tools/test/test_depsolve.py

```diff
@@ -1275,93 +1350,148 @@ def test_gen_config_combos(items_count, expected_combos):
+            {"name": "baseos", "address": "file:///baseos"},
+            {"name": "broken", "address": "file:///broken"},
+        ]
+    ),
+    (
+        {
+            "enabled_repos": [],
+            "additional_servers": [{"name": "broken", "address": "file:///broken"}]
+        },
+        [{"name": "baseos", "address": "file:///baseos"}, {"name": "custom", "address": "file:///custom"}],
+        [
+            {"name": "broken", "address": "file:///broken"},
+        ]
+    ),
+])
+def test_get_test_case_repo_servers(test_case, repo_servers, expected):
+    assert get_test_case_repo_servers(test_case, repo_servers) == expected
 
 
-# pylint: disable=too-many-branches
-@pytest.mark.parametrize("test_case", depsolve_test_cases, ids=tcase_idfn)
-@pytest.mark.parametrize("with_sbom", [False, True])
 @pytest.mark.parametrize("dnf_config, detect_fn", [
     ({}, assert_dnf),
     ({"use_dnf5": False}, assert_dnf),
     ({"use_dnf5": True}, assert_dnf5),
 ], ids=["no-config", "dnf4", "dnf5"])
-def test_depsolve(tmp_path, repo_servers, dnf_config, detect_fn, with_sbom, test_case):
+def test_depsolve_config_combos(tmp_path, repo_servers, dnf_config, detect_fn):
+    """
+    Test all possible configurations of repository configurations for the depsolve function.
+    Test on a single test case which installs two packages from two repositories.
+    """
     try:
         detect_fn()
     except RuntimeError as e:
         pytest.skip(str(e))
 
-    # pylint: disable=fixme
-    # TODO: remove this once dnf5 implementation is fixed
-    dnf5_broken_test_cases = [
-        "basic_pkg_group_with_excludes",
-        "install_pkg_excluded_in_another_transaction",
-        "error_pkg_not_in_enabled_repos",
-        "basic_module",
-    ]
-
-    if dnf_config.get("use_dnf5", False) and test_case["id"] in dnf5_broken_test_cases:
-        pytest.skip("This test case is known to be broken with dnf5")
-
+    test_case = depsolve_test_case_basic_2pkgs_2repos
     transactions = test_case["transactions"]
+    tc_repo_servers = get_test_case_repo_servers(test_case, repo_servers)
 
-    repo_servers_copy = repo_servers.copy()
-
-    # filter to only include enabled repositories
-    repo_servers_copy = [r for r in repo_servers_copy if r["name"] in test_case["enabled_repos"]]
-
-    if "additional_servers" in test_case:
-        repo_servers_copy.extend(test_case["additional_servers"])
-
-    for repo_configs, root_dir, opt_metadata in config_combos(tmp_path, repo_servers_copy):
+    for repo_configs, root_dir, opt_metadata in config_combos(tmp_path, tc_repo_servers):
         with TemporaryDirectory() as cache_dir:
-            res, exit_code = depsolve(transactions, repo_configs, root_dir,
-                                      cache_dir, dnf_config, opt_metadata, with_sbom)
+            res, exit_code = depsolve(
+                transactions, cache_dir, dnf_config, repo_configs, root_dir, opt_metadata)
 
             if test_case.get("error", False):
```

Comment:
AIUI this test now only runs `depsolve_test_case_basic_2pkgs_2repos` which does not test the "error" case, is the plan to extend this test or could this code path be removed?